zebra-consensus/src/checkpoint/tests.rs

```diff
@@ -213,63 +213,56 @@ async fn multi_item_checkpoint_list() -> Result<(), Report> {
+    }
+    for height in 0..=10 {
+        continuous_blockchain(Some(block::Height(height)), Testnet).await?;
     }
     Ok(())
 }
 
-/// Test a continuous blockchain, restarting verification at `restart_height`.
-// TODO: does this duplicate the test code commented out in src/chain/tests.rs?
-#[allow(dead_code)]
+/// Test a continuous blockchain on `network`, restarting verification at `restart_height`.
 #[spandoc::spandoc]
-async fn continuous_blockchain(restart_height: Option<block::Height>) -> Result<(), Report> {
+async fn continuous_blockchain(
+    restart_height: Option<block::Height>,
+    network: Network,
+) -> Result<(), Report> {
     zebra_test::init();
 
     // A continuous blockchain
-    let mut blockchain = Vec::new();
-    for b in &[
-        &zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_1_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_2_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_3_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_4_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_5_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_6_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_7_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_8_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_9_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_10_BYTES[..],
-    ] {
-        let block = Arc::<Block>::zcash_deserialize(*b)?;
-        let hash = block.hash();
-        blockchain.push((block.clone(), block.coinbase_height().unwrap(), hash));
-    }
-
-    // Parse only some blocks as checkpoints
-    let mut checkpoints = Vec::new();
-    for b in &[
-        &zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_5_BYTES[..],
-        &zebra_test::vectors::BLOCK_MAINNET_9_BYTES[..],
-    ] {
-        let block = Arc::<Block>::zcash_deserialize(*b)?;
-        let hash = block.hash();
-        checkpoints.push((block.clone(), block.coinbase_height().unwrap(), hash));
-    }
+    let blockchain = match network {
+        Mainnet => zebra_test::vectors::CONTINUOUS_MAINNET_BLOCKS.iter(),
+        Testnet => zebra_test::vectors::CONTINUOUS_TESTNET_BLOCKS.iter(),
+    };
+    let blockchain: Vec<_> = blockchain
+        .map(|(height, b)| {
+            let block = Arc::<Block>::zcash_deserialize(*b).unwrap();
+            let hash = block.hash();
+            let coinbase_height = block.coinbase_height().unwrap();
+            assert_eq!(*height, coinbase_height.0);
+            (block, coinbase_height, hash)
+        })
+        .collect();
+    let blockchain_len = blockchain.len();
 
-    // The checkpoint list will contain only block 0, 5 and 9
-    let checkpoint_list: BTreeMap<block::Height, block::Hash> = checkpoints
+    // Use some of the blocks as checkpoints
+    let checkpoint_list = vec![&blockchain[0], &blockchain[5], &blockchain[9]];
```

Comment:
What about using something as:

```
let checkpoint_list = vec![&blockchain[0], &blockchain[(blockchain_len-1)/2], &blockchain[blockchain_len-1]];
```

Reply:
Yeah, that's a nice tweak that would basically implement #1836.